urlce7759b8da_Palachsuper_EBA4_tgz-pJ8-json_TestJ8/src/terminal/tools/MCrypt.java

```diff
@@ -0,0 +1,131 @@
+/*
+mcrypt = new MCrypt();
+// Encrypt
+String encrypted = MCrypt.bytesToHex( mcrypt.encrypt("Text to Encrypt") );
+// Decrypt
+String decrypted = new String( mcrypt.decrypt( encrypted ) );
+*/
+
+package terminal.tools;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.NoSuchAlgorithmException;
+
+public class MCrypt {
+
+    private String iv = "361hr740823xy936";
+    private IvParameterSpec ivspec;
+    private SecretKeySpec keyspec;
+    private Cipher cipher;
+
+    public MCrypt(String SecretKey)
+    {
+        ivspec = new IvParameterSpec(iv.getBytes());
+
+        keyspec = new SecretKeySpec(SecretKey.getBytes(), "AES");
+
+        try {
+            cipher = Cipher.getInstance("AES/CBC/NoPadding");
+        } catch (NoSuchAlgorithmException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } catch (NoSuchPaddingException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
+
+    public byte[] encrypt(String text) throws Exception
+    {
+        if(text == null || text.length() == 0)
+            throw new Exception("Empty string");
+
+        byte[] encrypted = null;
+
+        try {
+            cipher.init(Cipher.ENCRYPT_MODE, keyspec, ivspec);
+
+            encrypted = cipher.doFinal(padString(text).getBytes());
+        } catch (Exception e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)